library/core/src/mem/maybe_uninit.rs

```diff
@@ -4,11 +4,15 @@ use crate::{fmt, intrinsics, ptr, slice};
 
 /// A wrapper type to construct uninitialized instances of `T`.
 ///
+/// A `MaybeUninit<T>` is like a `T`, but without the requirement that it is properly initialized as a `T`.
+/// Dropping a `MaybeUninit<T>` does nothing, even if properly initialized as a `T`, because
+/// the compiler relies on the type system to decide how to drop variables. Thus, if a `MaybeUninit<T>`
+/// should be dropped like a `T`, it should be converted to a `T` with `assume_init` or similar.
+///
 /// # Initialization invariant
 ///
-/// The compiler, in general, assumes that a variable is properly initialized
-/// according to the requirements of the variable's type. For example, a variable of
-/// reference type must be aligned and non-null. This is an invariant that must
+/// Every variable must be properly initialized according to the requirements of its type.
+/// For example, a variable of reference type must be aligned and non-null. This is an invariant that must
```

Comment:
You're still dong changes here without explaining the rhyme or reason for them. What is wrong with the sentence that you changed here?

Sorry, I'm already swamped with reviews, I really don't have the time to tease out the meaningful changes from the gratuitous rewordings in a PR like this.

Reply:
The idea here is to remove the distraction of mentioning the compiler. (Since it is really the language definition that matters, not what the compiler assumes...)

Reply:
That is the rigorous way of talking about it, but not the most accessible way. We have to be careful not to turn the docs into unreadable (to most of the audience) legalese.